### PR TITLE
Cortx 31873

### PR DESCRIPTION
**Pull request by VikasKumarC**
## Problem Statement

-   Master workload config design for default values

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [x]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [ ]  New/Affected tests are executed
-   [ ]  Attach test execution logs

## Review 

  Checklist for Author

-   [ ]  JIRA number/GitHub Issue added to PR
-   [ ]  PR is self reviewed
-   [ ]  Jira and state/status is updated and JIRA is updated with PR link
-   [ ]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [ ]  Changes done to ReadMe

-----
[View rendered docs/YAML_documents/yaml_structure.md](https://github.com/VikasKumarC/corio/blob/CORTX-31873/docs/YAML_documents/yaml_structure.md)

### CODE REVIEW

File: workload/master_config.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+s3api:
+  # TEST_ID: TEST-35748
+  object_size:
+    start: 0Kib
+    end: 100Kib
+  part_range: #Multipart Specific
+    start: 50
+    end: 100
+  min_runtime_to_pass: 2h
```
Comment (DhobaleVishal) [inline workload/master_config.yaml:9]: Are we planning to use this across all workload.

**Review by DhobaleVishal — APPROVED** [review]

File: workload/master_config.yaml (hunk 1 above)
Comment (VikasKumarC) [reply workload/master_config.yaml:9]: No, Wherever needed as this is multipart specific.

File: workload/master_config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+s3api:
```
Comment (kulrk) [inline workload/master_config.yaml]: This s3api should not be similar to any operation instead can we have a different name something like basic_io?

**Review by kulrk — APPROVED** [review]

File: workload/master_config.yaml (hunk 1 above)
Comment (priyankaborawake) [reply workload/master_config.yaml:9]: min_runtime_to_pass this change has to be made all over the config files.

File: workload/master_config.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    start: 0Kib
+    end: 100Kib
+  part_range: #Multipart Specific
+    start: 50
+    end: 100
+  min_runtime_to_pass: 2h
+  sessions_per_node: 1
+  range_read: 100bytes
+  tool: s3api
+  operation: bucket/object_range_read/object_fix_size/object_random_size/copy_object/multipart/multipart_partcopy
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: Should the operation be specified using / ? Or should it be list.
 Also if someone missed adding operation in config, which of above operation will be preformed by default.

File: workload/master_config.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    start: 0bytes
+    end: 1Gib
+  write_percentage: 100  # Percentage of data to fill the storage.
+  read_percentage: 100  # Percentage of data to be used to read from storage.
+  delete_percentage: 0  # Percentage of data to be deleted from storage.
+  cleanup_percentage: 0  # Percentage of data once reached then cleanup all data.
+  #  If None then it will be fetched from cluster(cortx specific.)
+  total_storage_size: None  # Total available storage from cluster or user to be used.
+  min_runtime_to_pass: 30d  # Minimum execution durations to mark workload pass.
+  sessions_per_node: 20  # Number of sessions per node.
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: Can we keep sessions_per_node 1 by default.

File: workload/master_config.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  background_delete: False  # Enable/Disable background delete(Cortx specific).
+  sample_size: 10000 # int (any user defined numbers) # samples should be user configurable like 10000
+  tool: s3api/s3bench
+  operation: "mix_object_ops"
+
+degraded_io:
+  pod_downtime: 20s #time taken to degraded pods (IO sleep/halt time), shutdown time #i.e. 20s
+  pod_downtime_schedule: 2h #time when nodes/pods need to be degraded [Time Delta in (HH:MM:SS)], before this IO will be in happy path
+  schedule_frequency: 1 #once or repetitive (after every schedule time interval) --> repetition will happen as per schedule time interval
+  pod_uptime_schedule: 2  #time when pods need to be active back since downtime. # when pods need to be reverted online[Time Delta in (HH:MM:SS) ]
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: Pod uptime schedule should also accept none to support permanent failure.

File: workload/master_config.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  inflight_mode: on/off #(Continuous IO once support is there for server handles the degraded nodes/pods)
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: I suggest to keep one value as default. 
all the other option can be mentioned in comment.

File: workload/master_config.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  ignore_io_failures : True/False #(IO failure may be ignored while IO continue with POD shutdown), Default is False
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: same for these.

**Review by priyankaborawake — CHANGES_REQUESTED** [review]
Added few comments.

File: workload/master_config.yaml (hunk 1 above)
Comment (VikasKumarC) [reply workload/master_config.yaml:9]: Yes, we will do once start implementing/considering master config in next phase

File: workload/master_config.yaml (hunk 2 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Agreed, Let me out in comment, we will think over and consider once will start implementing master_config.

File: workload/master_config.yaml (hunk 3 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: That was for reference, Let me change, I am keeping default value as bucket, and keeping all other operations in comments for user reference

File: workload/master_config.yaml (hunk 4 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Sure, Updated accordingly.

File: workload/master_config.yaml (hunk 5 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Ok, Mentioned and will have support for same.

File: workload/master_config.yaml (hunk 6 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Ok, Updating accordingly

File: workload/master_config.yaml (hunk 7 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Correct, Done

File: workload/master_config.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  read_percentage: 100  # Percentage of data to be used to read from storage.
+  delete_percentage: 0  # Percentage of data to be deleted from storage.
+  cleanup_percentage: 0  # Percentage of data once reached then cleanup all data.
+  #  If None then it will be fetched from cluster(cortx specific.)
+  total_storage_size: None  # Total available storage from cluster or user to be used.
+  min_runtime_to_pass: 30d  # Minimum execution durations to mark workload pass.
+  sessions_per_node: 1  # Number of sessions per node.
+  background_delete: False  # Enable/Disable background delete(Cortx specific).
+  sample_size: 10000 # int (any user defined numbers) # samples should be user configurable like 10000
+  tool: s3api/s3bench
```
Comment (priyankaborawake) [inline workload/master_config.yaml]: Please the options in comments.

**Review by priyankaborawake — APPROVED** [review]
Added minor comment. Approving PR.

File: workload/master_config.yaml (hunk 8 above)
Comment (VikasKumarC) [reply workload/master_config.yaml]: Done, Thanks

**Review by akshaym99 — APPROVED** [review]

**Review by rahul-telawade — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Pull request merged**